runtime/src/tasks/fifo/tasks-fifo.c

```diff
@@ -721,7 +717,13 @@ void chpl_task_executeTasksInList(void** p_task_list_void) {
                            child_ptask->id,
                            child_ptask->is_executeOn);
 
-    (*task_to_run_fun)(child_ptask->arg);
+    if (child_ptask->countRunningTasks)
```

Comment:
It's not a "while I was here" change. I moved the logic to adjust the counters from a wrapper task function (taskCallWrapper) into the main scheduler, so that I did not have to concern myself with further bundling/wrapping. Note that the current version in the branch with FIFO doesn't actually use the runtime area available for the task layer part of the bundle.